application/client-test/test/oasis-test.js

```diff
@@ -0,0 +1,52 @@
+const assert = require('assert');
+const config = require('../../config');
+const oasis = require('@oasislabs/client');
+const process = require('process');
+
+let bytecode = require('fs').readFileSync(config.WASM);
+
+const chains = {
+  local: {
+    gateway: new oasis.gateways.Web3Gateway(
+      config.WEB3_GATEWAY_URL,
+      oasis.Wallet.fromMnemonic(config.MNEMONIC)
+    ),
+    completion: chain => chain.gateway.disconnect(),
+  },
+  devnet: {
+    gateway: new oasis.gateways.Gateway(
+      config.DEVELOPER_GATEWAY_URL,
+    ),
+    completion: chain => {},
+  }
+};
+
+if (process.env.CHAIN === undefined) {
+  console.log("Please specify environment variable CHAIN={local, devnet}");
+  process.exit();
+}
+
+let chain = chains[process.env.CHAIN];
+if (chain === undefined) {
+  console.log("The options for CHAIN are 'local' or 'devnet'");
+  process.exit();
+}
+
+console.log(chain);
```

Comment:
lines 6-35 contain an incredible amount of boilerplate. Can you work with @armaniferrante to roll this into `oasis.js`?

Reply:
Let's make an RFC for this. At first glance, we'd want this to work with the cli driving the flow and defining the workspace.

Reply:
The cli is external to mocha. You'd have to pass everything in as args or envs. I'm not too keen on canonizing this boilerplate if we can fix it in a not unreasonable amount of time. Nothing is worse than large cargo culted files.

Reply:
We should enumerate some alternative ways of doing this before immediately copying the truffle workflow.